src/mds/inode_backtrace.cc

```diff
@@ -145,18 +145,7 @@ int inode_backtrace_t::compare(const inode_backtrace_t& other,
     if (ancestors[i].dirino != other.ancestors[i].dirino ||
         ancestors[i].dname != other.ancestors[i].dname) {
       *equivalent = false;
-      if (ancestors[i-1].version < other.ancestors[i-1].version) {
-        if (comparator > 0)
-          *divergent = true;
-        return -1;
-      } else if (ancestors[i-1].version > other.ancestors[i-1].version) {
-        if (comparator < 0)
-          *divergent = true;
-        return 1;
-      } else {
-        assert(ancestors[i-1].version == other.ancestors[i-1].version);
-        return 0;
-      }
```

Comment:
Removing this block is incorrect. You're right that this *should* be unreachable code if everything in the system is behaving, but we are trying to detect corruptions and bugs here. We need to detect it as divergent if we somehow have a newer ancestor version on the side with the older parent.